DataPlotly/core/plot_factory.py

```diff
@@ -278,6 +279,14 @@ def add_source_field_or_expression(field_or_expression):
             self.settings.data_defined_marker_sizes = marker_sizes
         if colors:
             self.settings.data_defined_colors = colors
+            try:
+                total = 1.0 / (len(colors) - 1)
+            except ZeroDivisionError:
+                total = 1.0
+            for current, color in enumerate(colors):
```

Comment:
You'd need to do this from the source color ramp itself, not from the evaluated colors (which may be in any order). You should be able to get this by:

1. Check if a color ramp transformer is set on the property. If it's not, then we probably don't want a legend anyway OR we'd need a totally different approach for the legend (e.g. if categorised colors are being used)

2. Loop through the colours in that ramp, and convert those and their corresponding positions to the plotly ramp format.

3. From the property's transformer you can also get the min and max value range, so you'd scale the numeric values in the plotly ramp to fit between these ranges. That should give the correct values in the legend.